core/extraction/utils.py

```diff
@@ -89,16 +89,19 @@ def datetime_excel_to_pandas(excel_dates_column: pd.Series) -> pd.Series:
     return python_dates_column
 
 
-def extract_postcodes(s: str | float) -> list[str]:
+def extract_postcodes(s: str | float) -> list[str] | None:
```

Comment:
sorry missed this one before, does this function actually take a float? That seems like a weird thing to have in there, maybe it's a hang-over from some old implementation? Could you check, and if this is nto the case, remove the union with float?

If it is the case then maybe we should re-think the function design, but I'm pretty certain it's a mistake